portfolio/src/main/webapp/script.js

```diff
@@ -17,12 +17,77 @@
  */
 function addRandomFact() {
   const facts =
-      ['I have a pet dog', 'I like to code'];
+      ['I have a pet dog!', 'I like to code!', 'Marquees are my favourite HTML element!'];
 
   // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
+}
+
+/**
+ * Controls slideshow image display on the page
+ */
+const deckNums = { // Enumerate slide decks
+  DOG_DECK: 0,
+  PLACE_DECK: 1,
+  FOOD_DECK: 2,
+};
+
+const PREV_SLIDE_DIR = -1;
+const NEXT_SLIDE_DIR = 1;
+
+class Slideshow {
```

Comment:
I like this refactor. I have a thought about a way to take it a bit
further. What if you accept the slideshow DOM element as an argument
to the constructor? Then you can add the onclick callbacks in the
constructor and avoid some duplicate code below.